### ISSUE TITLE
`vue/no-unused-properties` should handle `inject`

### ISSUE THREAD

**Issue by adamsol** [vuejs/eslint-plugin-vue#3035]
**What rule do you want to change?**
[vue/no-unused-properties](https://eslint.vuejs.org/rules/no-unused-properties)

**Does this change cause the rule to produce more or fewer warnings?**
More.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Allow `inject` as a new element in `groups`.

**Please provide some example code that this change will affect:**

```vue
<script>
    export default {
        inject: ['unused'],
    };
</script>
```

**What does the rule currently do for this code?**
Nothing.

**What will the rule do after it's changed?**
`warning  'unused' of inject found, but never used  vue/no-unused-properties`

**Additional context**
n/a
